config/env/all.js

```diff
@@ -13,5 +13,6 @@ module.exports = {
     cookieSecret: "session_cookie_secret_key_here",
     cryptoKey: "a_secure_key_for_crypto_here",
     cryptoAlgo: "aes256",
-    hostName: "localhost"
+    hostName: "localhost",
+    environmentalScripts: ["<script src='//localhost:35729/livereload.js'></script>"]
```

Comment:
Previously the host name in script tag was derived as `location.host || "localhost"`. I believe this was done to support development on cloud based IDEs where hostname is not localhost. Do you see any issues if we retain the same logic, instead of hardcoding it to localhost?

Reply:
Any suggestions on how to defer the evaluation of of `location` until it hits the browser?

Reply:
I guess we could have the `environmentalScripts` as an array that looks something like the following:

```
[{
  scriptTag: '<script src="//\`{{script.substitutions.loc}}\`:35729/livereload.js"></script>',
  substitutions: { loc: '${location.host || "localhost"}' } // No backticks so no eval.
}, {
  // Future scripts
}]
```

I'm not quite sure on the best way to get the `environmentalScripts` into the view yet, would need to play and see what's possible with the current templating lib. Ultimatly the `script.substitutions.loc` needs to slot into the scriptTag.

It's quite a bit of extra work, but what ever needs to be done.

Plausable @ckarande ?

Reply:
We could use the same trick as the original livereload script block, ie make the environmentalScripts line:
```
environmentalScripts: [`<script>document.write("<script src='http://" + (location.host || "localhost").split(":")[0] + ":35729/livereload.js'></" + "script>");</script>`]
```
The document.write happens on the client, injecting a script tag with the correct livereload.js URL.

Reply:
Ah yip, good thinking @rcowsill, I was over thinking it. Are you happy with Rob's suggestion @ckarande ? If so I'll make that tweak.

Reply:
yes. perfect.